pandas/core/groupby/groupby.py

```diff
@@ -1623,11 +1625,17 @@ def ohlc(self) -> DataFrame:
 
     @doc(DataFrame.describe)
     def describe(self, **kwargs):
-        with _group_selection_context(self):
-            result = self.apply(lambda x: x.describe(**kwargs))
-            if self.axis == 1:
-                return result.T
-            return result.unstack()
+        result = self.apply(lambda x: x.describe(**kwargs))
+        if self.axis == 1:
+            return result.T
+        # GH 34656 self.apply() will return non-nuisance grouping columns, but
+        # we remove them from describe if as_index=True
```

Comment:
why are we doing this compatiblity? this seems like a step backwards

Reply:
The current behaviour of `GroupBy.describe` is superior. It doesn't make sense to return a non-nuisance grouping column in the output of `GroupBy.describe()` because that column will, by construction, only have one value in each group and returning the *mean, std, min, 25%, 50%, 75%, max* of a single value is basically meaningless. 

If the user calls `GroupBy.describe` with `as_index=False` they will get a non-nuisance grouping column in the output with its mostly meaningless summary stats; but I think it make sense to leave these out by default (which is the current behaviour). 

The goal of this PR is just to fix the bug in `GroupBy.apply`, leaving the behaviour of `GroupBy.describe` (and any other function) unchanged.

Reply:
Following this update the `.describe` method is no longer affected by this PR. 

https://github.com/pandas-dev/pandas/pull/35314#issuecomment-664193224